File: app/src/main/java/com/example/desafioandroid/feature/pulls/PullViewModelFactory.java
package com.example.desafioandroid.feature.pulls;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class PullViewModelFactory implements ViewModelProvider.Factory {

    private Application mApplication;
    private String user, repository;

    public PullViewModelFactory(Application mApplication, String user, String repository) {
        this.mApplication = mApplication;
        this.user = user;
        this.repository = repository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new PullViewModel(mApplication, user, repository);
    }
}

File: app/src/main/java/com/joaoibarra/gitapp/features/pull/PullViewModelFactory.kt
package com.joaoibarra.gitapp.features.pull

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import com.joaoibarra.gitapp.features.repository.PullViewModel

class PullViewModelFactory (private val user: String, private val repo: String) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return  PullViewModel(user, repo) as T
    }

}